src/Core/Util/DateTime/NullDateTime.php

```diff
@@ -0,0 +1,279 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Util\DateTime;
+
+use DateTimeImmutable;
+use DateTimeInterface;
+use DateTimeZone;
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime as DateTimeUtil;
+use RuntimeException;
+
+/**
+ * Reflects null value of DateTime. Only format() method should be used, other methods might produce unexpected results
+ */
+class NullDateTime extends DateTimeImmutable
```

Comment:
Could you add unit test on this class ?

Reply:
Agreed on this one, we introduced this "exotic" class to fill our need for an empty value of DateTime but it could have unexpected behaviour if is wrongly used So we need to ensure that it will always be used for specific use cases and the other ones will throw exceptions I think only unit test will prevent it from being wrongly modified in the future